WMIAdventure/backend/WMIAdventure_backend/battle/businesslogic/effects/StopPlayerEffect.py

```diff
@@ -0,0 +1,22 @@
+from battle.businesslogic.effects.Effect import Effect
+from battle.businesslogic.player_states.StoppedPlayerState import StoppedPlayerState
+from cards.models import CardLevelEffects
+
+
+class StopPlayerEffect(Effect):
+    """
+    Stops target player for given amount of turns.
+    """
+
+    def __init__(self, effect_model: CardLevelEffects, turns_stopped: int=1):
+        """
+
+        :param effect_model:
+        :param turns_stopped: For how many turns player should be stopped.
+        """
+
+        super().__init__(effect_model)
+        self.turns_stopped = turns_stopped
+
+    def on_activation(self, target, turns_queue):
+        target.add_state(StoppedPlayerState(self.turns_stopped))
```

Comment:
W diagramach celowo umieściliśmy to turns_queue żeby móc go modyfikować w efektach. Moim zdaniem za zatrzymywanie ruchów playera powinna być odpowiedzialna właśnie ta kolejka ruchów. Implementowałem praktycznie to samo w podwajaniu kart [3b260b0](https://github.com/emkarcinos/WMIAdventure/commit/3b260b021796d6314b1ab6fa3e3b32873e93e5ff), tutaj wykorzystałem Deck żeby się tym zajął i nie musiałem niczego zmieniać w kartach. To tylko moja sugestia żeby zrobić taki sam mechanizm w TurnsQueue, ale możesz mnie spróbować przekonać do swojego rozwiązania

Reply:
Racja, działanie na kolejce byłoby lepsze. Plus tego rozwiązania jest taki, że umożliwia zatrzymanie playera na wiecej niz jedną turę, ale nie wiem czy jest nam takie coś potrzebne.